File: src/com/leetcode/69-sqrtx/69-sqrtx.java
package com.leetcode

class Solution {
    public int mySqrt(int x) {
       long start=1;
        long end=Integer.MAX_VALUE;
        long sqrt=1;
        if(x==0)
        return 0;
        while(start<=end){
            long mid=(start+end)/2;
          if(mid*mid<=x)
            {
                start=mid+1;
                sqrt=mid;}
            else end=mid-1;
        }
        
    return (int)sqrt;
    }
}
// binary search approach

File: src/com/leetcode/69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x)
    {
       long int i=1;
    while(i*i<=x)
    {
        if(i*i==x)
        {
            return i;
            break;
        }
        i++;
    }
        return i-1;
    }
};